runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -337,13 +371,18 @@ export class PageControllerBase {
         ? redirectTo(request, h, startPage)
         : redirectTo(request, h, `/${this.model.basePath}${startPage}`);
       this.setFeedbackDetails(viewModel, request);
+
+      //Calculate our relevantState, which will filter out previously input answers that are no longer relevant to this user journey
+      let relevantState = this.getConditionEvaluationContext(this.model, state);
```

Comment:
This is good, but it's only using the `relevantState` to filter out which components should be displayed. When the form is posted the next page is the journey is still being found using the full state object, so the bug is still present (see `makePostHandler`)

Reply:
Hi Max, I presume you mean makePostRouteHandler? 

I may be wrong but I don't believe we need to amend the state in the POST - The GET will always get called once a POST is complete in order to load the next page.

If you do believe this is an issue, are you able to provide us with a working example? (ie. the form json and the repro steps)

Reply:
Hi @MCrawleyHomeOffice, I can't chat today but have time at 10am tomorrow if needed. What I'm seeing is the post (`makePostRouteHandler`) when complete calls `return this.proceed(request, h, savedState);` where savedState is the full state including any now irrelevant paths. proceed then calls `getNextPage` to find the next page to redirect the user to. This does ultimately call the get handler, but for the new page that now needs to be rendered as the post/proceed call has already decided which page to go to next.

Reply:
This makes sense. Thanks, leave this with me for now whilst I investigate further.

Reply:
Slight update, we've managed to replicate the issue and fix it by working out the relevant state as per @superafroman 's suggestion. It's a tricky one to replicate due to the way we use the first "next" we find on a page that passes according to the required conditions, so we've updated the test form we use to smoke test this. Will add a smoke test around this to ensure we're checking for issues in the future.